app/models/tag.rb

```diff
@@ -40,6 +106,18 @@ def delete_and_update!
         tag_name:       name
       }
     )
-    destroy
+
+    # Create the delete activity.
+    repository.create_activity(
+      :delete,
+      owner:      actor,
+      recipient:  self,
+      parameters: {
+        repository_name: repository.name,
+        namespace_id:    repository.namespace.id,
```

Comment:
Well, whenever we propose removing namespaces, we would have to work around it in the same way I've done for removed images/tags: put the info in the parameters hash and update the recipient/tracker to point to a parent (e.g. in this case the namespace containing said image).